runtime/common/xwalk_resource_delegate.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+#define XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "base/macros.h"
+#include "base/memory/ref_counted_memory.h"
+#include "ui/base/resource/resource_bundle.h"
+
+namespace base {
+class FilePath;
+}
+
+namespace gfx {
+class Image;
+}
+
+namespace xwalk {
+
+// In Chrome,the locales path is set as "locales/" which potentially conflicts
+// with it is in Crosswalk. So purpose of XWalkResourceDelegate is to set the
+// locale pack path of Crosswalk as "locales/xwalk/".
+class XWalkResourceDelegate : public ui::ResourceBundle::Delegate {
+ public:
+  // Returns the singleton of delegate. It doesn't create an instance.
```

Comment:
Yes you are right, it is not referenced anywhere.  When I "copy" this part code from the src/chromecast, I noticed in cast_resource_delegate of the chromecast, the getinstance() is defined but not referenced either.  
My understanding is:  Since it is designed as a singleton pattern by using the g_instance member,  It is good to provide interface to get this instance. Although it is true we don't reference it so far. That's why I didn't remove it.